src/lib.rs

```diff
@@ -22,6 +22,9 @@ extern crate tracing;
 #[macro_use]
 extern crate derivative;
 
+#[cfg(all(feature = "tikv-jemallocator", feature = "mimalloc"))]
+compile_error!("features `tikv-jemallocator` and `mimalloc` are mutually exclusive");
```

Comment:
You might find it easier to have `mimalloc` take precedence in this case because the default featureset on *nix platforms includes `tikv-jemallocator` so it'll be difficult to override.

Reply:
> take precedence in this case

Please forgive my ignorance, I am not completely sure what `take precedence` means here. I gave it another try, could you please check it out? Thanks.